### PR TITLE
fixes #1960 Allow assessment completion to be excluded from display

### PR DESCRIPTION
**Pull request by dancgray**
Minimal change to allow control of how the assessment is included in _subProgressComplete.

Hopefully the intention is clearer. Assessments without _includeInPageProgress default to include pass/fail status in the completion calculation.

### CODE REVIEW

File: js/assessment.js [hunk 1]
```diff
@@ -212,10 +212,10 @@ define([
                 for (var i = 0, l = assessments.length; i < l; i++) {
                     var assessmentState = assessments[i].getState();
 
-                    if (assessmentState.includeInTotalScore && !assessmentState.isPass) continue;
+                    if (assessmentState.includeInPageProgress && assessmentState.includeInTotalScore && !assessmentState.isPass) continue;
```
Comment (oliverfoster) [inline js/assessment.js]: Did you need to decrement the...
```js
var assessmentsTotal = assessments.length;
```
value according to the number of assessments that are included not just passed>

Comment (dancgray) [reply js/assessment.js]: Depends if you think that the code is clearer than incrementing assessmentsPassed when includeInPageProgress is false and assessmentState.isComplete is true.

Comment (oliverfoster) [reply js/assessment.js]: https://github.com/adaptlearning/adapt-contrib-assessment/blob/8ce40146a0081fee3e40c9bfa246b5ec45bdcb09/js/assessment.js#L209-L228

I means in the sense that the total will still pull through. So despite having chosen not to include the assessment in the plp, at the moment the code is excluding it from the passed count but not the assessment count. To exclude it from plp you'll need to exclude it from both the total count and the passed count. Not just the passed count.

https://github.com/adaptlearning/adapt-contrib-pageLevelProgress/blob/master/js/completionCalculations.js#L40-L41

**Comment by oliverfoster** [comment]
is she still open?

**Comment by moloko** [comment]
This has been resolved in [PLP v.3.1.0](https://github.com/adaptlearning/adapt-contrib-pageLevelProgress/releases/tag/v3.1.0) with the new `_excludeAssessments` setting, hasn't it?

**Pull request closed without merging**